csrc/executor.cpp

```diff
@@ -366,13 +369,15 @@ void FusionExecutor::compileFusion(
   TORCH_INTERNAL_ASSERT(
       fusion_id_ > 0, "failed to assign a fusion_id_ after compilation.");
 
-  // The driver API call requires an int argument.
-  int max_dynamic_smem = 0;
-  CUDA_SAFE_CALL(cuFuncGetAttribute(
-      &max_dynamic_smem,
-      CU_FUNC_ATTRIBUTE_MAX_DYNAMIC_SHARED_SIZE_BYTES,
-      compiled_kernel_.function));
-  maybe_available_dynamic_smem_ = max_dynamic_smem;
+  // These should be nullopt at this point, but reset just in case
+  available_dynamic_smem_size_.reset();
```

Comment:
IIUC, `available_dynamic_smem_size_` and `static_smem_size_` should always be cleared together when we want to re-compute smem??

nitpick: maybe put them inside a helper function `reset_smem`.

Reply:
Yes. Will add the func